src/plugins/skills/src/systems/skill_handle_to_skill.rs

```diff
@@ -0,0 +1,267 @@
+use crate::{skills::Skill, traits::TryMap};
+use bevy::{
+	asset::{AssetEvent, Assets, Handle},
+	ecs::system::Res,
+	prelude::{Commands, Component, DetectChanges, Entity, EventReader, Query, Ref},
+};
+use common::{
+	errors::{Error, Level},
+	traits::try_insert_on::TryInsertOn,
+};
+
+pub(crate) fn skill_handle_to_skill<
+	TSource: Component + TryMap<Handle<Skill>, Skill, TResult>,
+	TResult: Component,
+>(
+	mut commands: Commands,
+	mut events: EventReader<AssetEvent<Skill>>,
+	skills: Res<Assets<Skill>>,
+	sources: Query<(Entity, Ref<TSource>)>,
+) -> Vec<Result<(), Error>> {
+	let force_update = events.read().any(added_or_modified);
+	let mut errors = vec![];
+
+	for (entity, source) in sources.iter().filter(is_changed_or(force_update)) {
```

Comment:
remove source to reduce component clutter.

Reply:
Removal breaks breaks functionality. For instance: remap skills, once skill assets change.